common/lib/dependabot/file_fetchers/base.rb

```diff
@@ -51,6 +51,24 @@ class Base
       GIT_SUBMODULE_CLONE_ERROR =
         /^fatal: clone of '(?<url>.*)' into submodule path '.*' failed$/
       GIT_SUBMODULE_ERROR_REGEX = /(#{GIT_SUBMODULE_INACCESSIBLE_ERROR})|(#{GIT_SUBMODULE_CLONE_ERROR})/
+      GIT_RETRYABLE_ERRORS =
+        T.let(
+          [
+            /remote error: Internal Server Error/,
+            /fatal: Couldn\'t find remote ref/,
```

Comment:
are these all guaranteed retryable errors? Some seem to be terminal errors and maybe we shouldn't retry on them. 

On the flip side, maybe a max 3 min retry interval is not so expensive?

Reply:
I agree, it's hard to say whether or not all of these are retryable without retrying them. If we merge this, and telemetry shows no noticeable difference, we can re-evaluate?

In researching this, I did find that there is significant overlap with a tool the Chromium team uses[^1].

[^1]: https://chromium.googlesource.com/infra/infra/+/HEAD/go/src/infra/tools/git/retry_regexp.go#17

Reply:
Makes sense; agree with you. And thanks for the Chromium link too!